File: Programmers/dfs1.js
const dfs = (numbers, target, index, value) => {
  if (index === numbers.length) {
    if (target === value) {
      return 1;
    } else {
      return 0;
    }
  }
  return (
    dfs(numbers, target, index + 1, value + numbers[index]) +
    dfs(numbers, target, index + 1, value - numbers[index])
  );
};

function solution(numbers, target) {
  var answer = dfs(numbers, target, 0, 0);
  return answer;
}

File: Programmers/dfs1.py
def dfs(tickets, route, visited):
    if len(route) == len(tickets) + 1:
        return route

    ret = []
    for i, ticket in enumerate(tickets):
        if not visited[i] and ticket[0] == route[-1]:
            visited[i] = True
            ret = dfs(tickets, route + [ticket[1]], visited)
            if len(ret) == len(tickets) + 1:
                return ret
            visited[i] = False
    return route


def solution(tickets):
    answer = []
    visited = [False] * len(tickets)
    tickets.sort()
    answer = dfs(tickets, ["ICN"], visited)
    return answer